presto-orc/src/main/java/com/facebook/presto/orc/metadata/MetadataReader.java

```diff
@@ -30,7 +33,12 @@
     PostScript readPostScript(byte[] data, int offset, int length)
             throws IOException;
 
-    Metadata readMetadata(HiveWriterVersion hiveWriterVersion, InputStream inputStream)
+    Metadata readMetadata(HiveWriterVersion hiveWriterVersion,
```

Comment:
I needed to move metadata decompression from the OrcReader to the MetadataReader, otherwise OrcReader would attempt to decompress entire DWRF metadata cache which is not right.